pkg/admission/validator_test.go

```diff
@@ -0,0 +1,231 @@
+									Path: v1alpha1.HTTPPathMatch{
+										Type:  "Prefix",
+										Value: "/",
+									},
+								},
+							},
+							Filters: []v1alpha1.HTTPRouteFilter{
+								{
+									Type: "RequestMirror",
+									RequestMirror: &v1alpha1.HTTPRequestMirrorFilter{
+										ServiceName: &testService,
+										Port:        8080,
+									},
+								},
+								{
+									Type: "RequestMirror",
+									RequestMirror: &v1alpha1.HTTPRequestMirrorFilter{
+										ServiceName: &specialService,
+										Port:        8081,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      false,
+			wantMessage: "HTTPRules cannot contain more than one HTTPFilter",
+			wantErr:     false,
+		},
+		{
+			name: "invalid httpRoute with 2 different filters",
+			hRoute: v1alpha1.HTTPRoute{
+				Spec: v1alpha1.HTTPRouteSpec{
+					Rules: []v1alpha1.HTTPRouteRule{
+						{
+							Matches: []v1alpha1.HTTPRouteMatch{
+								{
+									Path: v1alpha1.HTTPPathMatch{
+										Type:  "Prefix",
+										Value: "/",
+									},
+								},
+							},
+							Filters: []v1alpha1.HTTPRouteFilter{
+								{
+									Type: "RequestMirror",
+									RequestMirror: &v1alpha1.HTTPRequestMirrorFilter{
+										ServiceName: &testService,
+										Port:        8080,
+									},
+								},
+								{
+									Type: "RequestHeaderModifier",
+									RequestHeaderModifier: &v1alpha1.HTTPRequestHeaderFilter{
+										Add: map[string]string{"my-header": "bar"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      false,
+			wantMessage: "HTTPRules cannot contain more than one HTTPFilter",
+			wantErr:     false,
+		},
+		{
+			name: "invalid httpRoute with mix of filters and one duplicate",
+			hRoute: v1alpha1.HTTPRoute{
+				Spec: v1alpha1.HTTPRouteSpec{
+					Rules: []v1alpha1.HTTPRouteRule{
+						{
+							Matches: []v1alpha1.HTTPRouteMatch{
+								{
+									Path: v1alpha1.HTTPPathMatch{
+										Type:  "Prefix",
+										Value: "/",
+									},
+								},
+							},
+							Filters: []v1alpha1.HTTPRouteFilter{
+								{
+									Type: "RequestHeaderModifier",
+									RequestHeaderModifier: &v1alpha1.HTTPRequestHeaderFilter{
+										Set: map[string]string{"special-header": "foo"},
+									},
+								},
+								{
+									Type: "RequestMirror",
+									RequestMirror: &v1alpha1.HTTPRequestMirrorFilter{
+										ServiceName: &testService,
+										Port:        8080,
+									},
+								},
+								{
+									Type: "RequestHeaderModifier",
+									RequestHeaderModifier: &v1alpha1.HTTPRequestHeaderFilter{
+										Add: map[string]string{"my-header": "bar"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      false,
+			wantMessage: "HTTPRules cannot contain more than one HTTPFilter",
+			wantErr:     false,
+		},
+	}
+	for _, tt := range tests {
+		// copy variable to avoid scope problems with ranges
+		tt := tt
```

Comment:
Is this still necessary after the restructuring?

Reply:
Yes golint still complains about the scope if I don't add this 

```
pkg/admission/validator_test.go:264:38: Using the variable on range scope `tt` in function literal (scopelint)
			ok, msg, err := ValidateHTTPRoute(tt.hRoute)
			                                  ^
pkg/admission/validator_test.go:265:23: Using the variable on range scope `tt` in function literal (scopelint)
			if (err != nil) != tt.wantErr {
			                   ^
pkg/admission/validator_test.go:266:65: Using the variable on range scope `tt` in function literal (scopelint)
				t.Errorf("ValidateHTTPRoute() error = %v, wantErr %v", err, tt.wantErr)
				                                                            ^

```